.github/scripts/test-project.sh

```diff
@@ -73,12 +73,18 @@ if [ -n "${FORCE_PRISMA_CLIENT_CUSTOM_OUTPUT+x}" ]; then
   echo "-----------------------------"
   echo ""
   echo "FORCE_PRISMA_CLIENT_CUSTOM_OUTPUT=$FORCE_PRISMA_CLIENT_CUSTOM_OUTPUT, executing commands to turn the project into a custom output project"
-  echo ""
-  find . -type f -name "*.js" ! -path "*/node_modules/*" ! -name "*test*" -print0 | xargs -0 -r sed -i "s/@prisma\/client/db/g"
-  find . -type f -name "*.mjs" ! -path "*/node_modules/*" ! -name "*test*" -print0 | xargs -0 -r sed -i "s/@prisma\/client/db/g"
+
+  # whether the project has a src folder or not, we need to modify the path to the prisma client and how deep the path is
+  find . -mindepth 4 -type f -name "*.js" ! -path "*/node_modules/*" ! -name "*test*" -print0 | xargs -0 -r sed -i "s/@prisma\/client/\.\.\/\.\.\/\.\.\/prisma\/client/g"
+  find . -mindepth 4 -type f -name "*.mjs" ! -path "*/node_modules/*" ! -name "*test*" -print0 | xargs -0 -r sed -i "s/@prisma\/client/\.\.\/\.\.\/\.\.\/prisma\/client/g"
+  find . -mindepth 3 -type f -name "*.js" ! -path "*/node_modules/*" ! -name "*test*" -print0 | xargs -0 -r sed -i "s/@prisma\/client/\.\.\/\.\.\/prisma\/client/g"
+  find . -mindepth 3 -type f -name "*.mjs" ! -path "*/node_modules/*" ! -name "*test*" -print0 | xargs -0 -r sed -i "s/@prisma\/client/\.\.\/\.\.\/prisma\/client/g"
+  find . -mindepth 2 -type f -name "*.js" ! -path "*/node_modules/*" ! -name "*test*" -print0 | xargs -0 -r sed -i "s/@prisma\/client/\.\.\/prisma\/client/g"
+  find . -mindepth 2 -type f -name "*.mjs" ! -path "*/node_modules/*" ! -name "*test*" -print0 | xargs -0 -r sed -i "s/@prisma\/client/\.\.\/prisma\/client/g"
+  find . -mindepth 1 -type f -name "*.js" ! -path "*/node_modules/*" ! -name "*test*" -print0 | xargs -0 -r sed -i "s/@prisma\/client/\.\/prisma\/client/g"
+  find . -mindepth 1 -type f -name "*.mjs" ! -path "*/node_modules/*" ! -name "*test*" -print0 | xargs -0 -r sed -i "s/@prisma\/client/\.\/prisma\/client/g"
```

Comment:
Ugly buuut understandable and easily scaled if need be. Alternatively, we could make it a look, but harder to understand imo

```sh
  for depth in {4..1}; do
    find . -mindepth $depth -type f -name "*.js" ! -path "*/node_modules/*" ! -name "*test*" -print0 | xargs -0 -r sed -i "s/@prisma\/client/$(printf '../%.0s' $(seq 1 $((depth-1))))prisma\/client/g"
    find . -mindepth $depth -type f -name "*.mjs" ! -path "*/node_modules/*" ! -name "*test*" -print0 | xargs -0 -r sed -i "s/@prisma\/client/$(printf '../%.0s' $(seq 1 $((depth-1))))prisma\/client/g"
  done
  ```

Reply:
I think I find the loop easier to read honestly. They're still both rough on the line length tho, is it possible to separate them out over multiple? e.g.

```sh
find . -mindepth $depth -type f -name "*.js" 
    ! -path "*/node_modules/*" 
    ! -name "*test*" -print0 
    | xargs -0 -r sed -i "s/@prisma\/client/$(printf '../%.0s' $(seq 1 $((depth-1))))prisma\/client/g"
```

Reply:
I think at this point we might just as well use a proper programming instead of chain shell commands

Reply:
did just that